modules/features/filters/src/main/java/au/com/shiftyjelly/pocketcasts/playlists/edit/SmartPlaylistsOptionsFragment.kt

```diff
@@ -0,0 +1,173 @@
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@AndroidEntryPoint
+class SmartPlaylistsOptionsFragment : BaseDialogFragment() {
+    private val viewModel by viewModels<SmartPlaylistViewModel>({ requireParentFragment() })
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ) = content {
+        var isSelectingSortType by rememberSaveable { mutableStateOf(false) }
+        val performMediaRouteClick = remember { MutableSharedFlow<Unit>() }
+
+        val uiState by viewModel.uiState.collectAsState()
+        val playlist = uiState.smartPlaylist
+
+        DialogBox(
+            fillMaxHeight = false,
+            modifier = Modifier.nestedScroll(rememberNestedScrollInteropConnection()),
+        ) {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier.verticalScroll(rememberScrollState()),
+            ) {
+                Box(
+                    modifier = Modifier
+                        .padding(top = 8.dp)
+                        .background(MaterialTheme.theme.colors.primaryUi05, CircleShape)
+                        .size(56.dp, 4.dp),
+                )
+                if (playlist != null) {
+                    AnimatedContent(
+                        targetState = isSelectingSortType,
+                    ) { isSorting ->
+                        if (isSorting) {
+                            SmartPlaylistSortOptionsColumn(
+                                selectedSortType = playlist.episodeSortType,
+                                onSelectSortType = { type ->
+                                    viewModel.updateSortType(type)
+                                    dismiss()
+                                },
+                            )
+                        } else {
+                            SmartPlaylistOptionsColumn(
+                                sortType = playlist.episodeSortType,
+                                hasEpisodes = playlist.totalEpisodeCount > 0,
+                                onClickSelectAll = {
+                                    viewModel.startMultiSelecting()
+                                    dismiss()
+                                },
+                                onClickSortBy = { isSelectingSortType = true },
+                                onClickDownloadAll = {
+                                    downloadAll(playlist.totalEpisodeCount)
+                                    dismiss()
+                                },
+                                onClickChromecast = {
+                                    viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
The Chromecast menu option doesn't seem to work for me.


https://github.com/user-attachments/assets/de20ff3d-b345-4bb1-8991-8b3ab678d678